client/python/openlineage/client/transport/http.py

```diff
@@ -70,10 +83,544 @@ def create_token_provider(auth: dict[str, str]) -> TokenProvider:
+
+- **Deprecated parameters**: `session` and `adapter` parameters maintained with deprecation warnings
+- **Migration from requests**: Moved from requests library to httpx for better async support
+- **Configuration compatibility**: Existing HttpConfig remains compatible
+
+Example usage:
+```python
+# Async transport (recommended for high throughput)
+config = HttpConfig(
+    url="https://example.com",
+    async_config=AsyncConfig(enabled=True, max_concurrent_requests=50)
+)
+transport = HttpTransport(config)
+
+# Events are processed asynchronously with ordering guarantees
+transport.emit(start_event)      # Sent immediately
+transport.emit(complete_event)   # Waits for START success, then sent
+
+# Wait for all events to complete
+transport.wait_for_completion(timeout=30)
+
+# Get processing statistics
+stats = transport.get_stats()    # {"pending": 0, "success": 10, "failed": 0}
+
+# Graceful shutdown
+transport.shutdown(wait=True, timeout=30)
+
+# Sync transport (for simple use cases)
+config = HttpConfig(url="https://example.com")  # async disabled by default
+transport = HttpTransport(config)
+response = transport.emit(event)  # Blocks until sent, returns httpx.Response
+```
+"""
+
+
+class EventStatus:
+    PENDING = "pending"
+    SUCCESS = "success"
+    FAILED = "failed"
+
+
+class EventTracker:
+    def __init__(self):
+        self.events: dict[str, str] = {}
+        self.lock = threading.Lock()
+        # Queue for completion events waiting for their START event
+        self.pending_completion_events: dict[str, list] = {}  # run_id -> list of (event_id, body, headers)
+        self.stats = {
+            "pending": 0,
+            "success": 0,
+            "failed": 0,
+        }
+
+    def add_event(self, event_id: str):
+        with self.lock:
+            self.stats["pending"] += 1
+            self.events[event_id] = EventStatus.PENDING
+
+    def mark_success(self, event_id: str, run_id: str | None = None, event_type: str | None = None):
+        with self.lock:
+            self.stats["pending"] -= 1
+            self.stats["success"] += 1
+            if self.events.get(event_id) == EventStatus.PENDING:
+                del self.events[event_id]
+
+            # If this was a START event, mark it as successful and release any pending completion events
+            if run_id and event_type == "START":
+                # Return any pending completion events for this run
+                return self.pending_completion_events.pop(run_id, [])
+        return []
+
+    def mark_failed(self, event_id: str):
+        with self.lock:
+            log.debug("Marking event %s as failed", event_id)
+            self.stats["pending"] -= 1
+            self.stats["failed"] += 1
+            if self.events.get(event_id) == EventStatus.PENDING:
+                del self.events[event_id]
+
+    def add_pending_completion_event(
+        self, run_id: str, event_id: str, body: bytes | str, headers: dict[str, str]
```

Comment:
Maybe this set of arguments should be moved to a dedicated class Event or Request?